src/tests/testinput/3dvar_ice_obs.yaml

```diff
@@ -0,0 +1,103 @@
+cost function:
+  cost type: 3D-Var
+  time window:
+    begin: '2021-06-29T00:00:00Z'
+    length: P1D
+  analysis variables: [ice_area_fraction]
+  background error:
+    covariance model: SABER
+    saber central block:
+      saber block name: ID
+  geometry :
+    nemo variables:
+      - name: ice_area_fraction
+        nemo field name: iiceconc
+        model space: surface
+        field precision: double
+      - name: sea_surface_temperature
+        nemo field name: votemper
+        model space: surface
+        field precision: double
+      - name: sea_water_potential_temperature
+        nemo field name: votemper
+        model space: volume
+        field precision: double
+      - name: ice_area_fraction_background_error
+        nemo field name: sic_tot_var
+        model space: surface
+        variable type: background variance
+        field precision: double
+    grid name: ORCA2_T
+    number levels: 10
+  background :
+    date: 2021-06-29T12:00:00Z
+    state variables: [ ice_area_fraction ]
+    nemo field file: Data/orca2_t_nemo.nc
+  observations:
+    observers:
+    - obs space:
+        name: Sea Ice
+        obsdatain:
+          engine:
+            type: H5File
+            obsfile: Data/sic_obs_ideal.nc
+        obsdataout:
+          engine:
+            type: H5File
+            obsfile: testoutput/obsdataout_3dvar_ice.nc
+        simulated variables: [ice_area_fraction]
+      get values:
+        time interpolation: linear
+        atlas-interpolator:
+          type: unstructured-bilinear-lonlat
+          non_linear: missing-if-all-missing
+          max_fraction_elems_to_try: 0.0
+          adjoint: true
+      obs operator:
+        name: Composite
+        components:
+        - name: Identity
+      obs error:
+        covariance model: diagonal
+variational:
+  minimizer:
+    algorithm: DRIPCG
```

Comment:
Hi Daniel, I happened to have a quick glance at this out of curiosity after you messaged on the JEDI chat.
I noticed the `DRIPCG` minimizer being used. Just a heads-up that there's been some advice recently from JCSDA to avoid using this as "default", as it's apparently been shown to have occasional problems with convergence:
- Original OOPS issue JCSDA-internal/oops/issues/2554.
- Rick Rawlins has done some comparisons of `DRIPCG` vs `DRPCG` (the recommended alternative) for lfric-jedi, if interested: https://github.com/MetOffice/jedi-discussion/discussions/44. It shows pretty minimal impact in that case.

Reply:
Thanks very much @mo-joshuacolclough for the review! I hope you don't mind if we add you to the reviewer list on future PRs :) No pressure to review but good to try and get some eyes on from those who have already implemented this stuff!

Reply:
No problem! Yes more than happy to :)